lang/cs/Org.Apache.REEF.Common/Telemetry/CountersData.cs

```diff
@@ -0,0 +1,98 @@
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Linq;
+using Org.Apache.REEF.Tang.Annotations;
+using Org.Apache.REEF.Utilities.Logging;
+
+namespace Org.Apache.REEF.Common.Telemetry
+{
+    internal sealed class CountersData
+    {
+        private static readonly Logger Logger = Logger.GetLogger(typeof(CountersData));
+
+        /// <summary>
+        /// Registration of counters
+        /// </summary>
+        private readonly IDictionary<string, CounterData> _counterMap = new ConcurrentDictionary<string, CounterData>();
+
+        [Inject]
+        private CountersData()
+        {            
+        }
+
+        /// <summary>
+        /// Update counters 
+        /// </summary>
+        /// <param name="counters"></param>
+        internal void Update(ICounters counters)
+        {
+            foreach (var counter in counters.GetCounters())
+            {
+                CounterData counterData;
+                if (_counterMap.TryGetValue(counter.Name, out counterData))
+                {
+                    counterData.UpdateCounter(counter);
+                }
+                else
+                {
+                    _counterMap.Add(counter.Name, new CounterData(counter, counter.Value));
+                }
+
+                Logger.Log(Level.Verbose, "Counter name: {0}, value: {1}, description: {2}, time: {3},  incrementSinceLastSink: {4}.",
+                    counter.Name, counter.Value, counter.Description, new DateTime(counter.Timestamp), _counterMap[counter.Name].IncrementSinceLastSink);
+            }
+        }
+
+        /// <summary>
+        /// Reset increment since last sink for each counter
+        /// </summary>
+        internal void Reset()
+        {
+            foreach (var c in _counterMap.Values)
+            {
+                c.ResetSinceLastSink();
+            }
+        }
+
+        /// <summary>
+        /// Convert the counter data into ISet for sink
+        /// </summary>
+        /// <returns></returns>
+        internal ISet<KeyValuePair<string, string>> GetCounterData()
+        {
+            var set = new HashSet<KeyValuePair<string, string>>();
+            foreach (var c in _counterMap)
+            {
+                set.Add(c.Value.GetKeyValuePair());
+            }
+            return set;
+        }
+
+        /// <summary>
+        /// The condition that triggers the sink. The condition can be modified later.
+        /// </summary>
+        /// <returns></returns>
+        internal bool TriggerSink(int counterSinkThreshold)
+        {
+            return _counterMap.Values.Sum(e => e.IncrementSinceLastSink) > counterSinkThreshold;
```

Comment:
>Sum(e => e.IncrementSinceLastSink) [](start = 38, length = 34)

should we use absolute values here? 
if there are 2 counters going in different directions, net sum could stay close to 0